crates/sb_core/util/checksum.rs

```diff
@@ -2,17 +2,32 @@
 
 use ring::digest::Context;
 use ring::digest::SHA256;
+use std::fmt::Write;
 
 pub fn gen(v: &[impl AsRef<[u8]>]) -> String {
     let mut ctx = Context::new(&SHA256);
     for src in v {
         ctx.update(src.as_ref());
     }
     let digest = ctx.finish();
-    let out: Vec<String> = digest
+    let mut hash_str = String::with_capacity(64);
+    digest
         .as_ref()
         .iter()
-        .map(|byte| format!("{byte:02x}"))
-        .collect();
-    out.join("")
+        .for_each(|byte| write!(hash_str, "{byte:02x}")
+            .expect("write! macro on string cannot fail"));
```

Comment:
Let’s use faster_hex similar to the current Deno https://github.com/denoland/deno/blob/main/cli/util/checksum.rs

Reply:
faster_hex is fast!!. did a little benchmarking, for the three approaches of gen function.

1. using existing string concatenation
2. using write! macro
3. using faster_hex. 
here are the results:

|           | string concate | write! macro | faster-hex |
|-----------|----------------|--------------|------------|
| mean time | 2.791 us          | 973 ns        | 260 ns      |   

also attaching the violin plot for results

<img width="1021" alt="Screenshot 2024-04-17 at 6 44 18 PM" src="https://github.com/supabase/edge-runtime/assets/24789440/c689b69d-e4aa-425c-8c00-c4f9692b2869">

Reply:
Amazing thanks!